Tests/SwiftNavigationTests/IsolationTests.swift

```diff
@@ -3,34 +3,38 @@
   import XCTest
 
   class IsolationTests: XCTestCase {
-    @MainActor
-    func testIsolationOnMinActor() async {
-      let model = MainActorModel()
-      let expectation = expectation(description: "observation")
-      expectation.expectedFulfillmentCount = 2
-      let token = SwiftNavigation.observe {
-        _ = model.count
-        MainActor.assertIsolated()
-        expectation.fulfill()
+    func testIsolationOnMainActor() async throws {
+      try await Task { @MainActor in
```

Comment:
> For for synchronous tests, I do:
> 
> ```swift
> await MainActor.run {
>   …
> }
> ```

Since non-async tests run on the main thread in XCTest, can we simply omit this?

> Open to other ideas, or if there is some way to get Wasm to recognize `@MainActor` tests. The errors I got were at _runtime_, not compile, and looked like this:

I feel like I've seen similar on Linux before, where all test cases needed to be the same signature type for a test case to work, but I thought that was fixed...then again we don't have Linux tests on this repo, so it's possible this is broken in general for non-Apple platforms.

Reply:
> Since non-async tests run on the main thread in XCTest, can we simply omit this?

They run on the main thread, but they aren't known to be main actor to the compiler. So it won't play nicely with the isolation stuff we have for `observe`.